### DISCUSSION TITLE
Can't run post-successful build's APK

### DISCUSSION THREAD

**Discussion by AtanasNankinski** [codemagic-ci-cd/codemagic-docs#2444] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

- Finish build succesfully in either debug or release mode.
- Download the app-build.apk or app-release.apk file.
- Install directly to emulator by drag and drop or on physical device by runing the apk file on it.

### Expected results

After installing the APK file on the emulator/physical device to be able to run the app.

### Actual results

The app stays on white screen with the flutter logo.

### Build id (optional)

6605754533f8c319ec844f3d 6605705bb096a445e0ff28e7

**Comment by himesh-cm** [comment]
Hi @AtanasNankinski,

Could you kindly provide more information about the first screen? It's possible that if there are any dependencies on third-party services like Firebase or Bugsnag that are not configured properly, it could cause the screen to fail. Alternatively, you could set up error logging with a platform like Sentry or Raygun. 

Please let me know how it goes.

**Reply by AtanasNankinski** [reply]
I just started setting up the app and tried to see if the .apk files can be installed on device. There are no external services yet, just native splash screen followed by custom splash screen widget with delay and then another screen. Simple app at the begining of its development. It is being build okay on both emulators and physical device, but the .apk files provided by succesfull codemagic build can't run properly. Also on videos from 1-2 years ago i can see after successful build two .apk files are being provided - [app_name]_debug.apk and [app_name].apk but i get only [app_name]_debug.apk

**Reply by himesh-cm** [reply]
@AtanasNankinski, Can you please select `release` build within your `app > settings > Build > mode` and re-run the build.
Please test it on the release build and check if you still face any issues. Let me know how it goes.

**Reply by AtanasNankinski** [reply]
Already tried that, i tried all possible build options withing codemagic and has the same result.

**Comment by AtanasNankinski** [comment]
Well since it is clear i won't get help anytime soon i guess i will have to turn to other CI/CD services and leave Codemagic. Thank for the "amazing" support!

**Reply by himesh-cm** [reply]
Hello @AtanasNankinski,

I apologize for the delay in responding to your message. It seems like the issue you are facing is specific to your project. As per the logs, there seems to be a problem with compiling assets.

Here is the error message:

```
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
```

Could you please let me know what software configurations you are using locally, such as Flutter, java, etc.? I recommend keeping the same versions on Codemagic and trying again.

I see online a few options where using the lower gradle version could be helpful. Can you try it once on your end and let me know how it goes?

**Reply by AtanasNankinski** [reply]
The flutter version that i am using is 3.19.1 same as specified in the build options in codemagic. The gradle version is 7.6.1 and the java version from "java -version" command in the IDE Terminal is 21.0.1.

**Reply by himesh-cm** [reply]
Please try it with the same version on codemagic that you use locally and share the revised build ID if you are still facing any issues.

**Reply by himesh-cm** [reply]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
